### DISCUSSION TITLE
MacOS installation via ventoy

### DISCUSSION THREAD

**Discussion by rvvh472** [ventoy/Ventoy#1361] [Show and tell]
Hi
I've kabylake laptop And I've installed macos Monterey in it using dortania's guide AND made a Opencore UEFI bootable usb which successfully booted Monterey without any issue. After that I made an ISO file from the EFI folder using poweriso and pasted it my ventoyed usb stick. I was successful to run opencore EFi iso from ventoy that leads to the booting of macos Monterey. 

And I also clean installed the Monterey again using same OC EFI iso via ventoy and it was a success

Conclusion:
If we could just make a common OC EFI recovery for macos iso file, we'll be able to install macos via that iso file in ventoy. And if some how OC EFI functionality is enabled in ventoy via plugins it will be amazing

And if you design a script that will mount dmg and run .../MacOS/InstallAssistant in booted OC EFI recovery.iso this will leads to the gui macos installer

**Comment by flavi1** [comment]
But because we should be able to add/remove kexts, change OC config.plist, update OC...., putting OC in a common iso file is probably not the more useful way...
Another possible approche may be to reserve a path location to install OC on the big first Ventoy partition (but excluding it from ISOS files Parser in grub.conf). Then add automatically a menuentry to chainload OC at the bottom of Ventoy main menu if OC is detected. Or use it to load individuals dmg/macos iso parsed files... As suggested by rvvh472

I would like to do it, but i don't know how.
Maybe something like that at the end of grub.conf :

```
if [ -f $vtoy_iso_part/OC/OpenCore.efi ]; then
    menuentry "OpenCore detected" {
        echo -e "\n    PLACE OpenCore HERE ... "
		insmod chain
		insmod part_gpt
		#insmod fat
		insmod exfat
		set vtback_root=$root
		set root=$vtoy_iso_part
		chainloader $vtoy_iso_part/OC/OpenCore.efi
		set root=$vtback_root
    }
fi
```
Any ideas??

**Comment by flavi1** [comment]
I will simply try to chainload ventoy from OpenCore, by creating an OpenCore entry.
If it works, it would be good enought for me. But i think i will encounter a problem :
The EFI partition mini size.
Sub-question is : Is it possible to install OpenCore on the ISO partition (or another new one) instead of VTOYEFI ?
(Because we re unable to increase the 32MB EFI partition )
This is a crucial point. If it works, we will get a definitively complete full EFI bootloader asset (install, lives, rescues with grub ventoy & normal use with opencore as it now directly support linux...).
A one-for-all solution. Just because we may not want to care about it in the future...

**Comment by flavi1** [comment]
Or.....Another idea : 
Simply play with the already available ventoy module "file injection" to use with a common OC folder that contained specifics kexts and config.plist.
For convenience, does file injection module allows to put full folders (and sub folders contented inner)?
If it already does, the problem is just
1 - convert dmg to iso
2 - insert OC folder by the file injection module...
Any try?
I get no time to test it this month.
I assume it may be easy to realize a script able to do it...

**Reply by flavi1** [reply]
Oups... Impossible.
File injection is only supported by some linux distros! Not by OC!
:/

**Comment by roiplek** [comment]
It's clunky, but at the moment for me the only easily available solution is to use 2 USB drives, one with OC and the other with Ventoy, then chainload Ventoy after OC.

**Comment by halal-beef** [comment]
in terms of installing Mac OS I managed to make it work but it has some drawbacks, first create the usb in ventoy an then format the VENTOY partition (NOT VENTOYEFI) as fat32, u can then put ur efi folder and Mac OS install on the Ventoy partition then load OpenCore to then boot the installer and boot Mac OS, the only drawback is that it is fat32 so that means no other isos bigger than 4gb

**Reply by rvvh472** [reply]
This is possible.Here is Solution.
Copy your EFI to Ventoy Partition whether exfat, NTFS, fat32 doesn't matter. Create a custom Boot Entry in Ventoy Configuration to Opencore.efi. Here you go.

**Reply by sticky-note** [reply]
@rvvh472 Tried to do that, but when I select `EFI/OC/Opencore.efi`, I get
```
OC: Failed to load configuration!
```
Can you give some details of your custom Boot Entry in Ventoy for OpenCore, please ?

**Reply by hexman1999** [reply]
I could actually boot mac os installer by making img file of the whole efi partition which contains oc and adding a separate apfs partition contains the installer and is made on mac os at the end of flash drive 

Edit: that way I could have more than one installer and more than one OC version

**Reply by iwannet** [reply]
> I could actually boot mac os installer by making img file of the whole efi partition which contains oc and adding a separate apfs partition contains the installer and is made on mac os at the end of flash drive 
> 
> Edit: that way I could have more than one installer and more than one OC version 

Can you give a more detailed guide ?

**Reply by hexman1999** [reply]
> > I could actually boot mac os installer by making img file of the whole efi partition which contains oc and adding a separate apfs partition contains the installer and is made on mac os at the end of flash drive 
> > 
> > Edit: that way I could have more than one installer and more than one OC version 
> 
> Can you give a more detailed guide ?

The guide of creating mac os installer is included already in oc documentation, when you create ventoy keep space enough for the mac os installer partition and make img file by cloning efi partition with working oc then putting in ventoy directory

**Comment by roiplek** [comment]
> Edit: that way I could have more than one installer and more than one OC version

Other than quickly swapping out EFIs, there's no upside to doing it this way as this defeats Ventoys entire purpose

**Comment by StudentCyberOps** [comment]
@rvvh472 reading your text have been very educative for me, please answer this question of mine .
I can't install non Mac OS on Mac sometimes, what files do I  need to add? how can I establish it being influenced from your approach .

**Comment by JohnTvs09** [comment]
> Hi I've kabylake laptop And I've installed macos Monterey in it using dortania's guide AND made a Opencore UEFI bootable usb which successfully booted Monterey without any issue. After that I made an ISO file from the EFI folder using poweriso and pasted it my ventoyed usb stick. I was successful to run opencore EFi iso from ventoy that leads to the booting of macos Monterey.
> 
> And I also clean installed the Monterey again using same OC EFI iso via ventoy and it was a success
> 
> Conclusion: If we could just make a common OC EFI recovery for macos iso file, we'll be able to install macos via that iso file in ventoy. And if some how OC EFI functionality is enabled in ventoy via plugins it will be amazing
> 
> And if you design a script that will mount dmg and run .../MacOS/InstallAssistant in booted OC EFI recovery.iso this will leads to the gui macos installer

So... how do you do this on Ventoy? I am only familiar with IATKoS, iDeneb, and Niresh. I tried the Oralia method, but the GUI froze on me while I booted up the computer.

**Comment by GAURAV9615540348** [comment]
> Hi I've kabylake laptop And I've installed macos Monterey in it using dortania's guide AND made a Opencore UEFI bootable usb which successfully booted Monterey without any issue. After that I made an ISO file from the EFI folder using poweriso and pasted it my ventoyed usb stick. I was successful to run opencore EFi iso from ventoy that leads to the booting of macos Monterey.
> 
> And I also clean installed the Monterey again using same OC EFI iso via ventoy and it was a success
> 
> Conclusion: If we could just make a common OC EFI recovery for macos iso file, we'll be able to install macos via that iso file in ventoy. And if some how OC EFI functionality is enabled in ventoy via plugins it will be amazing
> 
> And if you design a script that will mount dmg and run .../MacOS/InstallAssistant in booted OC EFI recovery.iso this will leads to the gui macos installer

Please, can you explain.....
   Where did you put the main MacOS installation file (.dmg or .raw) ?
In the EFI.iso or alongside of EFI.iso ?
I can't able to contact with you. you are a genius, wants to discuss with you, if you want knock me at telegram @BARISH95
